### PR TITLE
perf(qwen3-14b prefill): reduce task fragmentation with SPMD kernels

### PR DESCRIPTION
**Pull request by Leaf-Salix**
## Summary
- Optimize Qwen3-14B prefill scheduling in `models/qwen3/14b/prefill_fwd.py`, reducing excessive fine-grained task dispatch while keeping the model interface and runner inputs unchanged.
- Rework the prefill layer task layout around RoPE/KV-cache, attention, projection, RMSNorm, MLP, and residual phases so the generated orchestration uses fewer, larger, and better-balanced tasks than the original fragmented task graph.
- Fix Q RoPE correctness in the optimized path so the optimized prefill remains numerically aligned with the original Qwen3-14B implementation.
- Golden PASS on a2a3 for the Qwen3-14B prefill validation path, including `batch=1 max_seq=128`.

| Scenario | Metric | Original / Revert | Optimized | Reduction | Speedup / Ratio |
|---|---|---:|---:|---:|---:|
| 40-layer `batch=1 max_seq=128` L2 swimlane, 3 runs | Mean task count | 351050 | 31706 | 91.0% | 11.1x fewer |
| 40-layer `batch=1 max_seq=128` L2 swimlane, 3 runs | Mean Total Test Time | 572.544 ms | 77.385 ms | 86.5% | 7.40x faster |
| Real-weight serving replay swimlane, 5 runs | Mean AICore task count | 333562 | 36434 | 89.1% | 9.16x fewer |
| Real-weight serving replay swimlane, 5 runs | Mean prefill device span | 579.371 ms | 76.967 ms | 86.7% | 7.53x faster |

- A rerun with the current PR head (`743ce75`) in the real-weight serving replay path produced the same top token (`" These"`) and improved the prefill L2 device span to ~77.0 ms.

## Notes
- This PR is focused on Qwen3-14B prefill task scheduling and correctness. It does not require pypto runtime changes.
- The timing numbers are L2 swimlane/device-side measurements. End-to-end serving wall time includes additional L3/runtime/dispatch/drain overhead and should be interpreted separately from the prefill kernel device span.
- The latest real-weight serving replay run produced the L2 records and logits successfully; the task cleanup path later reported an `asc_dumper` shared-library cleanup error, so only the emitted L2 device span and logits summary are used as evidence.

### CODE REVIEW

File: models/qwen3/14b/prefill_fwd.py [hunk 1]
```diff
@@ -79,16 +79,239 @@
...
+                                                    q_padded,
+                                                    k_tile,
+                                                    b_trans=True,
+                                                    out_dtype=pl.FP32,
+                                                )
+                                                raw_row0 = (
+                                                    gi * ATTN_TOK_GROUP * Q_HEAD_PAD
+                                                    + dd * Q_HEAD_PAD
+                                                    + sb * Q_HEAD_PAD
+                                                )
```
[the remark this answers is not in the document]
Comment (Leaf-Salix) [reply models/qwen3/14b/prefill_fwd.py]: Fixed in the current head. The attention scratch layout no longer encodes `sb` into the local row offset, and `_attention_phase_window` now carries online softmax state across sequence blocks via `prev_mi/prev_li/prev_oi -> mi_new/li_new/oi_new` before finalize. I also validated the optimized path with `batch=1 max_seq=256 --use-max-seq num_layers=1` golden PASS.

File: models/qwen3/14b/prefill_fwd.py [hunk 2]
```diff
@@ -79,16 +79,239 @@
...
+    pl.Tensor[[ATTN_TOK_GROUP * TOTAL_Q_GROUPS * Q_HEAD_PAD, HEAD_DIM], pl.FP32],
+]:
+    raw_scores_group = pl.create_tensor(
+        [ATTN_TOK_GROUP * TOTAL_Q_GROUPS * Q_HEAD_PAD, SEQ_TILE],
+        dtype=pl.FP32,
+    )
+    exp_padded_group = pl.create_tensor(
+        [ATTN_TOK_GROUP * TOTAL_Q_GROUPS * Q_HEAD_PAD, SEQ_TILE],
+        dtype=pl.BF16,
+    )
```
[the remark this answers is not in the document]
Comment (Leaf-Salix) [reply models/qwen3/14b/prefill_fwd.py]: Fixed in the current head. The optimized attention path now keeps per-phase local scratch for QK/softmax/SV and accumulates the online softmax state across `sb` blocks into `oi_tmp_phase` / `cur_li_phase` before finalize. This removes the old row aliasing issue and preserves full-prefix attention for `ctx_blocks > 1`. Validated with `batch=1 max_seq=256 --use-max-seq num_layers=1` golden PASS.

File: models/qwen3/14b/prefill_fwd.py [hunk 3]
```diff
@@ -146,417 +369,757 @@ def prefill_layer(
...
+                    for ob in pl.range(q_core, Q_OUT_BLOCKS, Q_PROJ_SPMD_BLOCKS):
+                        q0 = ob * Q_OUT_CHUNK
+                        tile_a = pl.slice(normed_tile, [TOK_TILE, K_CHUNK], [0, 0])
+                        tile_w = pl.slice(wq, [K_CHUNK, Q_OUT_CHUNK], [layer_hidden_base, q0])
+                        q_acc = pl.matmul(tile_a, tile_w, out_dtype=pl.FP32)
+                        for kb in pl.pipeline(1, HIDDEN_BLOCKS, stage=2):
+                            k0 = kb * K_CHUNK
+                            tile_a_i = pl.slice(normed_tile, [TOK_TILE, K_CHUNK], [0, k0])
+                            tile_w_i = pl.slice(wq, [K_CHUNK, Q_OUT_CHUNK], [layer_hidden_base + k0, q0])
+                            q_acc = pl.matmul_acc(q_acc, tile_a_i, tile_w_i)
```
[the remark this answers is not in the document]
Comment (Leaf-Salix) [reply models/qwen3/14b/prefill_fwd.py:564]: I tested this in a separate tuning experiment before changing the PR.

The full no-peel form across q_proj / kv_proj / out_proj / gate_proj / up_proj / down_proj failed to compile for down_proj with an unsupported `pto.tmov` address-space pair, likely due to the dynamic `ob_start` / `ob_end` pipeline range plus the `if ob == ob_start` initialization branch.

I also tested a partial version that keeps down_proj peeled and applies the branch-inside-pipeline form to the other projection stages. That version passed golden for both max_seq=128 and max_seq=256, but the 4-layer L2 swimlane did not show a stable performance improvement:

- current v225 reference: 3194 tasks, 8798.34 us, avg exec 64.21 us
- partial no-peel rerun: 3194 tasks, 9072.04 us, avg exec 65.96 us

Given the compile issue for down_proj and the lack of measured benefit for the remaining stages, I am leaving the current peeled pipeline structure unchanged in this PR.
